### ISSUE TITLE
operator-sdk olm status does not work with standard openshift installations

### ISSUE THREAD

**Issue by fgiloux** [operator-framework/operator-sdk#4714]
#### What did you do?

$ operator-sdk olm status --version 0.16.1 --olm-namespace openshift-operator-lifecycle-manager

#### What did you expect to see?

All listed resources having the status install as per the documentation: https://sdk.operatorframework.io/docs/olm-integration/quickstart-bundle/#enabling-olm

#### What did you see instead? Under which circumstances?
```
NAME                                            NAMESPACE    KIND                        STATUS
operators.operators.coreos.com                               CustomResourceDefinition    Installed
operatorgroups.operators.coreos.com                          CustomResourceDefinition    Installed
installplans.operators.coreos.com                            CustomResourceDefinition    Installed
clusterserviceversions.operators.coreos.com                  CustomResourceDefinition    Installed
subscriptions.operators.coreos.com                           CustomResourceDefinition    Installed
system:controller:operator-lifecycle-manager                 ClusterRole                 Installed
aggregate-olm-edit                                           ClusterRole                 Installed
aggregate-olm-view                                           ClusterRole                 Installed
catalogsources.operators.coreos.com                          CustomResourceDefinition    Installed
olm                                                          Namespace                   namespaces "olm" not found
olm-operator-binding-olm                                     ClusterRoleBinding          clusterrolebindings.rbac.authorization.k8s.io "olm-operator-binding-olm" not found
olm-operator                                    olm          Deployment                  deployments.apps "olm-operator" not found
catalog-operator                                olm          Deployment                  deployments.apps "catalog-operator" not found
olm-operator-serviceaccount                     olm          ServiceAccount              serviceaccounts "olm-operator-serviceaccount" not found
operators                                                    Namespace                   namespaces "operators" not found
global-operators                                operators    OperatorGroup               operatorgroups.operators.coreos.com "global-operators" not found
olm-operators                                   olm          OperatorGroup               operatorgroups.operators.coreos.com "olm-operators" not found
packageserver                                   olm          ClusterServiceVersion       clusterserviceversions.operators.coreos.com "packageserver" not found
operatorhubio-catalog                           olm          CatalogSource               catalogsources.operators.coreos.com "operatorhubio-catalog" not found
```
#### Environment

OpenShift 4.6 

**Operator type:**

/language go

**Kubernetes cluster type:**

OpenShift

`$ operator-sdk version`
`operator-sdk version: "v1.5.0", commit: "98f30d59ade2d911a7a8c76f0169a7de0dec37a0", kubernetes version: "1.19.4", go version: "go1.15.5", GOOS: "linux", GOARCH: "amd64"`

`$ go version`
`go version go1.15.6 linux/amd64`

`$ kubectl version`
`Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0+9f84db3", GitCommit:"9f84db336d1a77cba52684ecb51bfb197e9b4533", GitTreeState:"clean", BuildDate:"2020-10-30T09:33:51Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}`

#### Possible Solution

Part of the issue is that the namespace passed in the command line is not taken in consideration in GetObjectsStatus
https://github.com/operator-framework/operator-sdk/blob/master/internal/olm/client/status.go#L53

By fixing it I get:
```
NAME                                            NAMESPACE                               KIND                        STATUS
operators.operators.coreos.com                  openshift-operator-lifecycle-manager    CustomResourceDefinition    Installed
operatorgroups.operators.coreos.com             openshift-operator-lifecycle-manager    CustomResourceDefinition    Installed
installplans.operators.coreos.com               openshift-operator-lifecycle-manager    CustomResourceDefinition    Installed
clusterserviceversions.operators.coreos.com     openshift-operator-lifecycle-manager    CustomResourceDefinition    Installed
subscriptions.operators.coreos.com              openshift-operator-lifecycle-manager    CustomResourceDefinition    Installed
system:controller:operator-lifecycle-manager    openshift-operator-lifecycle-manager    ClusterRole                 Installed
packageserver                                   openshift-operator-lifecycle-manager    ClusterServiceVersion       Installed
aggregate-olm-view                              openshift-operator-lifecycle-manager    ClusterRole                 Installed
olm-operator                                    openshift-operator-lifecycle-manager    Deployment                  Installed
olm-operators                                   openshift-operator-lifecycle-manager    OperatorGroup               Installed
catalogsources.operators.coreos.com             openshift-operator-lifecycle-manager    CustomResourceDefinition    Installed
olm-operator-serviceaccount                     openshift-operator-lifecycle-manager    ServiceAccount              Installed
aggregate-olm-edit                              openshift-operator-lifecycle-manager    ClusterRole                 Installed
catalog-operator                                openshift-operator-lifecycle-manager    Deployment                  Installed
olm                                             openshift-operator-lifecycle-manager    Namespace                   namespaces "olm" not found
global-operators                                openshift-operator-lifecycle-manager    OperatorGroup               operatorgroups.operators.coreos.com "global-operators" not found
olm-operator-binding-olm                        openshift-operator-lifecycle-manager    ClusterRoleBinding          clusterrolebindings.rbac.authorization.k8s.io "olm-operator-binding-olm" not found
operators                                       openshift-operator-lifecycle-manager    Namespace                   namespaces "operators" not found
operatorhubio-catalog                           openshift-operator-lifecycle-manager    CatalogSource               catalogsources.operators.coreos.com "operatorhubio-catalog" not found
```
There are still issues:
- The namespaces resources defined in olm.yaml should not be used instead the names should be taken from the annotation of the packageserver csv in the target cluster: 
  - olm.operatorNamespace: openshift-operator-lifecycle-manager
  - olm.targetNamespaces: openshift-operator-lifecycle-manager
- The name of the OperatorGroup is also part of an annotation in the same csv:
  - olm.operatorGroup: olm-operators
- The name of the clusterrolebinding is based on the namespace, hence olm-operator-binding-openshift-operator-lifecycle-manager instead of olm-operator-binding-olm
- For the CatalogSource OpenShift comes with these sources as pre-installed:
openshift-marketplace/certified-operators
openshift-marketplace/community-operators
openshift-marketplace/redhat-marketplace
openshift-marketplace/redhat-operators
I am however not sure that it matters at all.

https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.17.0/olm.yaml

**Comment by estroz** [comment]
We should audit how `olm status` gets namespaces and names for resources, and change the command to look up objects as you suggest.

/kind bug

**Comment by kgibm** [comment]
Hi, I ran into this issue as I was about to test using OLM with a customer Go operator in my OpenShift cluster, and I'm not sure if this is a blocker or not. The [OLM Integration Bundle Tutorial](https://sdk.operatorframework.io/docs/olm-integration/tutorial-bundle/#enabling-olm) writes:

> All resources listed should have status Installed.

Can I still continue with the tutorial?

**Comment by VenkatRamaraju** [comment]
Currently, we install all resources from [olm.yaml](https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.19.1/olm.yaml), which is taken directly from the olm repo. While getting the statuses of the resources (``operator-sdk olm status``), the [getResource](https://github.com/VenkatRamaraju/operator-sdk/blob/namespace-filtering/internal/olm/installer/client.go#L159) function retrieves the resource list from the bindata that was compiled into the SDK during installation. 

This is where we see the ``Not found`` error in the olm status command. This is because olm is automatically installed in OpenShift clusters in a way that we don't really know much about.

Questions that came up during discussions - 
1. Do we continue to support ``olm status`` or use a different mechanism to determine whether olm is installed correctly in a cluster?
2. Could we detect the use of an OpenShift cluster and add a warning message?
3. Add to SDK docs that olm status is mostly geared towards Kubernetes clusters.

**Comment by fgiloux** [comment]
IMO 2. and 3. are not real options. Most of OLM users are also OpenShift users. If operator-sdk olm status has any value it should work independently of the way OLM has been installed.

**Comment by asmacdo** [comment]
Clearing milestone to re-triage

**Comment by varshaprasad96** [comment]
`operator-sdk olm` command is really handy for testing operators locally, we would still want to support the `status` command. As @fgiloux  mentioned, we may have to either find a way to refactor `olm status` to make sure it identifies the resources which consider its deployment successful in openshift and vanilla k8s clusters. Option 2, could be a temporary fix to inform users of the limitations of the command. To ensure complete support, we may have to identify the cluster type either way and verify if all the resources are present (if there is no other possibility).

**Comment by varshaprasad96** [comment]
Based on the discussion held during bug triage, the following points came up which need more discussion:
1. Should we support providing the `status` of olm in a cluster when it is not installed using `operator-sdk olm install` command?
2. Since there are various ways to install and verify if olm is present, would it be better if OLM provides a `status` api which can be used by us?
3. As a temporary fix, should we check if the cluster is OCP and accordingly modify our logic to check olm's status?

**Comment by jberkhahn** [comment]
Think we have to 2 things.
1. In the short term, add a blurb to https://sdk.operatorframework.io/docs/olm-integration/quickstart-bundle/#enabling-olm about how `olm status` works, that it only works with standard installations like from `olm install`, and how to manually verify it in the event of a nonstandard installation.
2. Talk with the OLM WG about how best to approach solving this problem - could they expose some sort of status API to keep this simple on our end? If we have investigate the cluster to determine this it could be messy and difficult to keep updated.

**Comment by jmrodri** [comment]
Fixed by PR #5373
